code/_helpers/game.dm

```diff
@@ -400,15 +400,17 @@
 * Gets the highest and lowest pressures from the tiles in cardinal directions
 * around us, then checks the difference.
 */
-/proc/getOPressureDifferential(var/turf/loc)
-	var/minp=16777216;
-	var/maxp=0;
+/proc/get_surrounding_pressure_differential(var/turf/loc)
+	var/minp = 16777216 // Arbitrary high value
+	var/maxp = 0
 	for(var/dir in global.cardinal)
-		var/turf/T = get_step(loc,dir)
-		var/datum/gas_mixture/environment = T.return_air()
-		var/cp = environment?.return_pressure()
-		if(cp<minp)minp=cp
-		if(cp>maxp)maxp=cp
+		var/turf/neighbor = get_step(loc,dir)
+		if(!neighbor || (neighbor.c_airblock(loc) & AIR_BLOCKED))
```

Comment:
I may be stupid. This needs to be done in a way that's at least a little smarter than I suggested, because firedoors block air, no? Or did you test this and it still worked?